src/ocamlorg_package/lib/ocamlorg_package.ml

```diff
@@ -202,14 +202,40 @@ let featured_packages t = t.featured_packages
 module Documentation = struct
   type toc = { title : string; href : string; children : toc list }
 
-  type item =
-    [ `Module of string
-    | `ModuleType of string
-    | `Parameter of int * string
-    | `Class of string
-    | `ClassType of string ]
-
-  type t = { toc : toc list; module_path : item list; content : string }
+  type breadcrumb_kind =
+    | Page
+    | LeafPage
+    | Module
+    | ModuleType
+    | Parameter of int
+    | Class
+    | ClassType
+    | File
+
+  let breadcrumb_kind_from_string s =
+    match s with
+    | "page" -> Page
+    | "leaf-page" -> LeafPage
+    | "module" -> Module
+    | "module-type" -> ModuleType
+    | "class" -> Class
+    | "class-type" -> ClassType
+    | "file" -> File
+    | _ ->
+        if String.starts_with ~prefix:"argument-" s then
+          let i = List.hd (List.tl (String.split_on_char '-' s)) in
+          Parameter (int_of_string i)
+        else raise (Invalid_argument ("kind not recognized: " ^ s))
```

Comment:
Isn't this almost the same as `Module_map.kind_of_json`? Isn't some sharing possible?

Reply:
After we update how voodoo generates `package.json`, which contains the module map, sharing makes sense. Before that, there are subtle differences and keeping them separate is the cleaner route.

Reply:
Sorry, I was being too hasty here, yeah, looks like they are now actually the same, except for that one fallback case. I tried not to do too many cleanups in this patch to keep the diff small. :see_no_evil:

Definitely looking forward to refactor and clean that up. Probably best to do it at the same time as removing the fallback, though.